src/sst/core/model/python/pymodel_unitalgebra.cc

```diff
@@ -0,0 +1,338 @@
+    nullptr,                // binaryfunc nb_or
+    SST_NB_COERCE                // coercion nb_coerce Python 2 only
+    SST_NB_INTLONG((unaryfunc)unitAlgebraToLong) // unaryfunc nb_int (py1 nb_ling for py2)
+    SST_NB_RESERVED         // nb_reserved, py3 only
+    nullptr,                // unaryfunc nb_float
+    SST_NB_OCT                // unaryfunc nb_oct
+    SST_NB_HEX                // unaryfunc nb_hex
+    
+    /* Added in release 2.0 */
+    (binaryfunc)unitAlgebraAdd,  // binaryfunc nb_inplace_add
+    (binaryfunc)unitAlgebraSub,  // binaryfunc nb_inplace_subtract
+    (binaryfunc)unitAlgebraMul,  // binaryfunc nb_inplace_multiply
+    SST_NB_INPLACE_DIVIDE((binaryfunc)unitAlgebraDiv)   // binaryfunc nb_inplace_divide, py2 only
+    nullptr,                // binaryfunc nb_inplace_remainder
+    nullptr,                // ternaryfunc nb_inplace_power
+    nullptr,                // binaryfunc nb_inplace_lshift
+    nullptr,                // binaryfunc nb_inplace_rshift
+    nullptr,                // binaryfunc nb_inplace_and
+    nullptr,                // binaryfunc nb_inplace_xor
+    nullptr,                // binaryfunc nb_inplace_or
+    
+    /* Added in release 2.2 */
+    nullptr,                // binaryfunc nb_floor_divide
+    (binaryfunc)unitAlgebraDiv, // binaryfunc nb_true_divide
+    nullptr,                // binaryfunc nb_inplace_floor_divide
+    (binaryfunc)unitAlgebraDiv,  // binaryfunc nb_inplace_true_divide
+    
+    /* Added in release 2.5 */
+    nullptr,                // unaryfunc nb_index
+    SST_NB_MATRIX_MULTIPLY     // py3 only
+    SST_NB_INPLACE_MATRIX_MULTIPLY // py3 only
+}; 
+    
+// Other methods
+
+static PyObject* unitAlgebraGetRoundedValue(PyObject *self, PyObject *UNUSED(args))
+{
+    return unitAlgebraToLong(self);
+}
+
+static PyObject* unitAlgebraHasUnits(PyObject* self, PyObject* args)
+{
+    char *units = NULL;
+        
+    if ( PyArg_ParseTuple(args, "s", &units) ) {
+        UnitAlgebraPy_t *ua = (UnitAlgebraPy_t*)self;
+        if ( ua->obj.hasUnits(units) ) Py_RETURN_TRUE;
+    }
+    Py_RETURN_FALSE;
+}
+
+static PyObject* unitAlgebraInvert(PyObject *self, PyObject *UNUSED(args))
+{
+    PyObject *argList = Py_BuildValue("(O)", self);
+    PyObject* ret = PyObject_CallObject((PyObject*)&PyModel_UnitAlgebraType, argList);    
+    Py_DECREF(argList);
+    UnitAlgebraPy_t *ret_ua = (UnitAlgebraPy_t*)ret;
+    ret_ua->obj.invert();
+    return ret;
+}
+
+
+static PyMethodDef unitAlgebraMethods[] = {
+    {   "getRoundedValue",
+        unitAlgebraGetRoundedValue, METH_NOARGS,
+        "Rounds value of UnitAlgebra to nearest whole number and returns a long"},
+    {   "hasUnits",
+        unitAlgebraHasUnits, METH_VARARGS,
+        "Checks to see if the UnitAlgebra has the specified units"},
+    {   "invert",
+        unitAlgebraInvert, METH_NOARGS,
+        "Inverts the UnitAlgebra value and units"},
+    // {   "str",
+    //     unitAlgebraStr, METH_NOARGS,
```

Comment:
What is the plan for this? Is there a reason it's commented?

Reply:
No, that's just a remnant from when I didn't know how to actually do the str function.  I'll make another pass through looking for commented out blocks to remove.